fbpic/particles/particles.py

```diff
@@ -180,6 +186,13 @@ def dens_func( z, r ) ...
         self.By = np.zeros( Ntot )
 
         self.use_envelope = use_envelope
+        if self.use_envelope:
+            self.a2 = np.zeros( Ntot )
+            self.grad_a2_z = np.zeros( Ntot )
+            self.grad_a2_x = np.zeros( Ntot )
+            self.grad_a2_y = np.zeros( Ntot )
```

Comment:
Could you add `dtype='complex'` here? This because these buffers are first used in order to gather the complex a field.

Reply:
OK, based on off-line discussions: this actually not needed because of recent changes in the way that the gather functions were written.